src/ops/ops.go

```diff
@@ -206,22 +212,35 @@ func (o *ops) PrepareController() error {
 }
 
 func (o *ops) renderControllerCm() error {
-
-	controllerDeployTemplate := filepath.Join(controllerDeployFolder, controllerDeployCmTemplate)
-	templateData, err := ioutil.ReadFile(controllerDeployTemplate)
-	if err != nil {
-		o.log.Errorf("Error occurred while trying to read %s : %e", controllerDeployTemplate, err)
-		return err
-	}
-	var assistedControllerParams = map[string]string{
+	var params = map[string]string{
 		"InventoryHost": config.GlobalConfig.Host,
 		"InventoryPort": fmt.Sprintf("\"%d\"", config.GlobalConfig.Port),
 		"ClusterId":     config.GlobalConfig.ClusterID,
 	}
-	o.log.Infof("Filling template file %s", controllerDeployTemplate)
+
+	return o.renderControllerDeploy(filepath.Join(controllerDeployFolder, controllerDeployCmTemplate),
+		params, renderedControllerCm)
+}
+
+func (o *ops) renderControllerPod() error {
+	var params = map[string]string{
+		"ControllerImage": config.GlobalConfig.ControllerImage,
+	}
+
+	return o.renderControllerDeploy(filepath.Join(controllerDeployFolder, controllerDeployPodTemplate),
+		params, renderedControllerPod)
+}
+
+func (o *ops) renderControllerDeploy(srcTemplate string, params map[string]string, dest string) error {
```

Comment:
Since this method is generic :+1: 
```suggestion
func (o *ops) renderDeploymentFiles(srcTemplate string, params map[string]string, dest string) error {
```